hw/ip/otbn/dv/otbnsim/sim/state.py

```diff
@@ -498,3 +512,6 @@ def stop_at_end_of_cycle(self, err_bits: int) -> None:
         '''
         self._err_bits |= err_bits
         self.pending_halt = True
+
+    def invalidate_imem(self) -> None:
+        self._time_to_imem_invalidation = 2
```

Comment:
You may be able to use a value of 1 here, as there's only a single instruction in the prefetch stage, however does depend upon the timing of when the value is checked and updated.

Reply:
No, this needs to be 2 (if I'd got it wrong in either direction, the test would have failed). It's one cycle for the memory latency, plus one cycle for the prefetch stage.